spec/unit/primitive_after_spec.rb

```diff
@@ -0,0 +1,103 @@
+require_relative '../../libraries/primitive_after'
+
+describe Choregraphie::After do
+  let(:block) { proc {} }
+  let(:choregraphie) do
+    Choregraphie::Choregraphie.new('test') do
+      after(&block)
+    end
+  end
+
+  describe 'standard behavior' do
+    before do
+      # all file manipulation are no-op
+      allow(FileUtils).to receive(:touch)
+      allow(FileUtils).to receive(:rm)
+    end
+
+    context 'when node has been installed long time ago' do
+      before do
+        allow(File).to receive(:exist?).with(/installed/).and_return(true)
+      end
+
+      context 'when a choregraphie is in progress' do
+        it 'must call its block' do
+          allow(File).to receive(:exist?).with(/inprogress/).and_return(true)
+          expect(block).to receive(:call)
+          choregraphie.cleanup.each(&:call)
+        end
+
+        it 'must mark its in progress state' do
+          expect(File).to receive(:exist?).with(/inprogress/).at_least(:once).and_return(false)
+          expect(FileUtils).to receive(:touch).with(/inprogress/)
+          choregraphie.before.each(&:call)
+        end
+      end
+
+      context 'when no choregraphie is in progress' do
+        it 'must not call its block' do
+          allow(File).to receive(:exist?).with(/inprogress/).and_return(false)
+          expect(block).not_to receive(:call)
+          choregraphie.cleanup.each(&:call)
+        end
+      end
+    end
+
+    context 'when node has just been installed' do
+      before do
+        allow(File).to receive(:exist?).with(/installed/).and_return(false)
+      end
+      context 'when a choregraphie is in progress' do
+        before { allow(File).to receive(:exist?).with(/inprogress/).and_return(true) }
+        it 'must call its block' do
+          expect(block).to receive(:call)
+          choregraphie.cleanup.each(&:call)
+        end
+        it 'must mark server as "installed"' do
+          expect(FileUtils).to receive(:touch).with(/installed/)
+          choregraphie.cleanup.each(&:call)
+        end
+      end
+
+      context 'when no choregraphie is marked in progress' do
+        before { allow(File).to receive(:exist?).with(/inprogress/).and_return(false) }
+        it 'must call its block' do
+          expect(block).to receive(:call)
+          choregraphie.cleanup.each(&:call)
+        end
+
+        it 'must mark server as "installed"' do
+          expect(FileUtils).to receive(:touch).with(/installed/)
+          choregraphie.cleanup.each(&:call)
+        end
+      end
```

Comment:
duplication of tests with the above context: you can factorise those two examples and use them in both contexts.

Reply:
indeed, will do